shotover-proxy/src/frame/cassandra.rs

```diff
@@ -92,18 +91,61 @@ pub mod raw_frame {
 pub struct CassandraMetadata {
     pub version: Version,
     pub stream_id: StreamId,
-    pub tracing_id: Option<Uuid>,
+    pub tracing: Tracing,
     pub opcode: Opcode,
-    pub flags: Flags,
     // missing `warnings` field because we are not using it currently
 }
 
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum Tracing {
+    Request(bool),
+    Response(Option<Uuid>),
+}
+
+impl Tracing {
+    fn enabled(&self) -> bool {
+        match self {
+            Self::Request(enabled) => *enabled,
+            Self::Response(uuid) => uuid.is_some(),
+        }
+    }
+}
+
+impl From<Tracing> for Option<Uuid> {
+    fn from(tracing: Tracing) -> Self {
+        match tracing {
+            Tracing::Request(_) => None,
+            Tracing::Response(uuid) => uuid,
+        }
+    }
+}
+
+impl Tracing {
+    fn from_frame(frame: &RawCassandraFrame) -> Result<Self> {
+        match frame.direction {
+            Direction::Request => Ok(Self::Request(frame.flags.contains(Flags::TRACING))),
+            Direction::Response => {
+                if frame.tracing_id.is_none() && frame.flags.contains(Flags::TRACING) {
```

Comment:
Cant we just rely on cassandra-protocol to give us sensible values?
Why do we need to check its work?